src/lite-yt-embed.js

```diff
@@ -11,37 +11,167 @@
-         * cannot rely on anything and have to use the YouTube Data API.
-         *
-         * amp-youtube also eschews using the API, so they just try sddefault with a hqdefault fallback:
-         *   https://github.com/ampproject/amphtml/blob/6039a6317325a8589586e72e4f98c047dbcbf7ba/extensions/amp-youtube/0.1/amp-youtube.js#L498-L537
-         * For now I'm gonna go with this confident (lol) assertion: https://stackoverflow.com/a/20542029, though I'll use `i.ytimg` to optimize for origin reuse.
-         *
-         * Worth noting that sddefault is _higher_ resolution than hqdefault. Naming is hard. ;)
-         * From my own testing, it appears that hqdefault is ALWAYS there sddefault is missing for ~10% of videos
-         *
-         * TODO: Do the sddefault->hqdefault fallback
-         *       - When doing this, apply referrerpolicy (https://github.com/ampproject/amphtml/pull/3940)
-         * TODO: Consider using webp if supported, falling back to jpg
-         */
-        this.posterUrl = `https://i.ytimg.com/vi/${this.videoId}/hqdefault.jpg`;
+        this.iframeNode = document.createElement('iframe');
+        this.iframeNode.setAttribute('frameborder', '0');
+        this.iframeNode.setAttribute('allow', 'accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture');
+        this.iframeNode.setAttribute('allowfullscreen', '');
+        this.iframeNode.setAttribute('src', this.embedSourceUrl);
+    }
+
+    /**
+     * Lo, the youtube placeholder image!  (aka the thumbnail, poster image, etc)
+     * There is much internet debate on the reliability of thumbnail URLs. Weak consensus is that you
+     * cannot rely on anything and have to use the YouTube Data API.
+     *
+     * amp-youtube also eschews using the API, so they just try sddefault with a hqdefault fallback:
+     *   https://github.com/ampproject/amphtml/blob/6039a6317325a8589586e72e4f98c047dbcbf7ba/extensions/amp-youtube/0.1/amp-youtube.js#L498-L537
+     * For now I'm gonna go with this confident (lol) assertion: https://stackoverflow.com/a/20542029, though I'll use `i.ytimg` to optimize for origin reuse.
+     *
+     * Worth noting that sddefault is _higher_ resolution than hqdefault. Naming is hard. ;)
+     * From my own testing, it appears that hqdefault is ALWAYS there sddefault is missing for ~10% of videos
+     *
+     * TODO: Do the sddefault->hqdefault fallback
+     *       - When doing this, apply referrerpolicy (https://github.com/ampproject/amphtml/pull/3940)
+     * TODO: Consider using webp if supported, falling back to jpg
+     *
+     * @returns {string}
+     */
+    get posterUrl() {
+        return `https://i.ytimg.com/vi/${this.videoId}/hqdefault.jpg`;
+    }
+
+
+    /**
+     * Retrieve the URL targeting the embed video from YouTube.
+     *
+     * @returns {string}
+     */
+    get embedSourceUrl() {
+        return `https://www.youtube-nocookie.com/embed/${this.videoId}?autoplay=1&${this.params}`;
```

Comment:
an alternative is to use `new URL()` and then `url.searchParams = this.params`

and at that point i'd probably probably put the `autoplay=1` bit into the params getter.

butttt on the other hand.. i dont think its terribly important to use the URL constructor here. we're dealing with a very constrained url.

Reply:
> we're dealing with a very constrained url.

My exact reason for not using it.

However, if based on some other issue discussion we end up allowing more than just single video embeds (ref #47 for example) then that kind of thing would open my mind up to using a URL object.